### PR TITLE
improve webserver conf mastercontainer

### PR DESCRIPTION
**Pull request by Zoey2936**

### CODE REVIEW

**Comment by szaimen** [comment]
@Zoey2936 thanks for this! May I ask why this change is needed? :)

File: Containers/mastercontainer/Caddyfile [hunk 1]
```diff
@@ -20,6 +20,13 @@ http://:80 {
   redir https://{host}{uri}
 }
 
+https://:8080 {
+        reverse_proxy localhost:8000
+        tls internal {
+                on_demand
```
Comment (szaimen) [inline Containers/mastercontainer/Caddyfile:26]: Will this still work also with ip-addresses now that https://github.com/nextcloud/all-in-one/blob/15da1035fb814c559b385219e1d992684e9075dc/Containers/mastercontainer/Caddyfile#L18-L20 is in place?

File: Containers/mastercontainer/start.sh [hunk 2]
```diff
@@ -261,36 +261,24 @@ chown www-data:www-data -R /mnt/docker-aio-config/session/
 chown www-data:www-data -R /mnt/docker-aio-config/caddy/
 chown root:root -R /mnt/docker-aio-config/certs/
 
+IPv4_ADDRESS_NC="$(dig nextcloud-aio-nextcloud A +short | grep '^[0-9.]\+$' | sort | head -n1)"
+IPv6_ADDRESS_NC="$(dig nextcloud-aio-nextcloud AAAA +short | grep '^[0-9a-fA-F:]\+$' | sort | head -n1)"
```
Comment (szaimen) [inline Containers/mastercontainer/start.sh:265]: this only works after the initial startup is complete and the nextcloud container is running and the mastercontainer was restarted at least once...

Comment (Zoey2936) [reply Containers/mastercontainer/start.sh:265]: ohhh... that is true...

File: Containers/mastercontainer/Caddyfile (hunk 1 above)
Comment (Zoey2936) [reply Containers/mastercontainer/Caddyfile:26]: not sure

Comment (Zoey2936) [reply Containers/mastercontainer/Caddyfile:26]: But I think internal tls ignores ask

File: Containers/mastercontainer/apache.conf [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ErrorLog /proc/self/fd/2
+
+LogLevel error
+
+<IfModule log_config_module>
+    LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" combined
+    LogFormat "%h %l %u %t \"%r\" %>s %b" common
+
+    <IfModule logio_module>
+      LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\" %I %O" combinedio
```
Comment (szaimen) [inline Containers/mastercontainer/apache.conf:46]: can we use the same logformat like below?

File: Containers/mastercontainer/Caddyfile (hunk 1 above)
Comment (szaimen) [reply Containers/mastercontainer/Caddyfile:26]: would need to be verified then...

**Comment by szaimen** [comment]
Also conflicts...

File: Containers/mastercontainer/Caddyfile (hunk 1 above)
Comment (DenuxPlays) [reply Containers/mastercontainer/Caddyfile:26]: > But I think internal tls ignores ask

Yes they do

File: Containers/mastercontainer/apache.conf [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    DirectoryIndex index.html
+</IfModule>
+
+<Files ".ht*">
+    Require all denied
+</Files>
+
+ErrorLog /proc/self/fd/2
+
+LogLevel error
```
Comment (szaimen) [inline Containers/mastercontainer/apache.conf:39]: We might switch to warn with that...
```suggestion
LogLevel warn
```

File: Containers/mastercontainer/apache.conf [hunk 5]
```diff
@@ -0,0 +1,106 @@
+StartServers             3
+MinSpareThreads         75
+MaxSpareThreads        250
+ThreadsPerChild         25
+MaxRequestWorkers      400
+MaxConnectionsPerChild   0
+PidFile "/run/apache2/httpd.pid"
```
Comment (szaimen) [inline Containers/mastercontainer/apache.conf:7]: I suppose these are the defaults?

File: Containers/mastercontainer/apache.conf [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+ServerTokens Prod
+ServerSignature On
```
Comment (szaimen) [inline Containers/mastercontainer/apache.conf:10]: what does this?

File: Containers/mastercontainer/apache.conf [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    AddType application/x-compress .Z
+    AddType application/x-gzip .gz .tgz
+</IfModule>
+
+<IfModule mime_magic_module>
+    MIMEMagicFile /etc/apache2/magic
+</IfModule>
+
+Include /etc/apache2/conf.d/conf.d/languages.conf
```
Comment (szaimen) [inline Containers/mastercontainer/apache.conf:67]: not sure about these?

File: Containers/mastercontainer/apache.conf (hunk 3 above)
Comment (Zoey2936) [reply Containers/mastercontainer/apache.conf:46]: just change it, how you need it

File: Containers/mastercontainer/Dockerfile [hunk 8]
```diff
@@ -74,49 +76,16 @@ RUN set -ex; \
...
-        echo "LoadModule ssl_module modules/mod_ssl.so" | tee -a /etc/apache2/httpd.conf; \
-        echo "LoadModule socache_shmcb_module modules/mod_socache_shmcb.so" | tee -a /etc/apache2/httpd.conf; \
-        echo "Include /etc/apache2/sites-available/mastercontainer.conf" | tee -a /etc/apache2/httpd.conf; \
-    \
-    rm -f /etc/apache2/conf.d/default.conf \
-          /etc/apache2/conf.d/userdir.conf \
-          /etc/apache2/conf.d/info.conf; \
-    \
-    rm -rf /var/www/localhost/cgi-bin/; \
+    mkdir /etc/apache2/logs; \
```
[the remark this answers is not in the document]
Comment (Zoey2936) [reply Containers/mastercontainer/Dockerfile:79]: I think yes

File: Containers/mastercontainer/apache.conf (hunk 5 above)
Comment (Zoey2936) [reply Containers/mastercontainer/apache.conf:7]: yes

File: Containers/mastercontainer/apache.conf [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+<IfModule log_config_module>
+    LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" combined
+    LogFormat "%h %l %u %t \"%r\" %>s %b" common
+
+    <IfModule logio_module>
+      LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\" %I %O" combinedio
+    </IfModule>
+
+    CustomLog /etc/apache2/logs/access.log combined
```
[the remark this answers is not in the document]
Comment (Zoey2936) [reply Containers/mastercontainer/apache.conf:49]: if you want, yes

File: Containers/mastercontainer/apache.conf (hunk 6 above)
Comment (Zoey2936) [reply Containers/mastercontainer/apache.conf:10]: grafik
grafik

File: Containers/mastercontainer/apache.conf (hunk 7 above)
Comment (Zoey2936) [reply Containers/mastercontainer/apache.conf:67]: just the default conf

Comment (szaimen) [reply Containers/mastercontainer/apache.conf:67]: do we need it?

Comment (Zoey2936) [reply Containers/mastercontainer/apache.conf:67]: I would keep it

Comment (szaimen) [reply Containers/mastercontainer/apache.conf:67]: ok

File: Containers/mastercontainer/apache.conf (hunk 6 above)
Comment (szaimen) [reply Containers/mastercontainer/apache.conf:10]: So do we need it?

Comment (Zoey2936) [reply Containers/mastercontainer/apache.conf:10]: servertokens yes, the other thing, not sure

**Comment by szaimen** [comment]
I guess I can close this?

**Pull request closed without merging**